### PR TITLE
chore: made a component file for docker image details

### PR DESCRIPTION
**Pull request by randhir-devtron**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                                </span>
+                            </span>
+                        </Tippy>
+                    </div>
+                )
+            })}
+            {deployment.dockerImages.length > 3 && (
+                <div onClick={() => setShowDockerInfo(true)} className="cursor anchor ml-10">
+                    <span>
+                        <span className="">{deployment.dockerImages.length - 3}</span>
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:33]: Why are we setting "" as className

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                        <span className="ml-3">more</span>
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:34]: if possible avoid margins

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import Tippy from '@tippyjs/react'
+import { ChartDeploymentDetail } from './chartDeploymentHistory.service'
+import docker from '../../../assets/icons/misc/docker.svg'
+
+const DockerImageDetails = ({
+    deployment,
+    setShowDockerInfo,
+}: {
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:9]: Move it to interface

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import Tippy from '@tippyjs/react'
+import { ChartDeploymentDetail } from './chartDeploymentHistory.service'
+import docker from '../../../assets/icons/misc/docker.svg'
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:4]: Check if we can import as ReactComponent

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:9]: And move it to separate file

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                                <span className="ml-3" data-testid="docker-version-deployment-history">
+                                    {dockerImage.split(':')[1] || dockerImage}
+                                </span>
+                            </span>
+                        </Tippy>
+                    </div>
+                )
+            })}
+            {deployment.dockerImages.length > 3 && (
+                <div onClick={() => setShowDockerInfo(true)} className="cursor anchor ml-10">
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:31]: use button if we are setting onClick

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import Tippy from '@tippyjs/react'
+import { DockerImageDetailsProps } from './chartDeploymentHistory.service'
+import { ReactComponent as DockerIcon } from '../../../assets/icons/misc/docker.svg'
+
+const DockerImageDetails = ({ deployment, setShowDockerInfo }: DockerImageDetailsProps) => {
+    const handleOnClick = () => {
+        setShowDockerInfo(true)
+    }
+    return (
+        <div>
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:11]: Can we use React Fragment

File: src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            {deployment.dockerImages.slice(0, 3).map((dockerImage, index) => {
```
Comment (AbhishekA1509) [inline src/components/v2/chartDeploymentHistory/DockerImageDetails.tsx:12]: @vivek-devtron will dockerImage be unique?

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request closed without merging**